core/src/main/scala/org/apache/spark/api/r/RUtils.scala

```diff
@@ -44,7 +44,7 @@ private[spark] object RUtils {
         (sys.props("spark.master"), sys.props("spark.submit.deployMode"))
       } else {
         val sparkConf = SparkEnv.get.conf
-        (sparkConf.get("spark.master"), sparkConf.get("spark.submit.deployMode"))
+        (sparkConf.get("spark.master"), sparkConf.get("spark.submit.deployMode", "cluster"))
```

Comment:
if anything the default deploy mode should be `client` to be consistent with assumptions in the code elsewhere